source/adapters/level_zero/event.cpp

```diff
@@ -1124,8 +1129,8 @@ ur_result_t ur_event_handle_t_::reset() {
 
   if (!isHostVisible())
     HostVisibleEvent = nullptr;
-
-  ZE2UR_CALL(zeEventHostReset, (ZeEvent));
+  if (!UrQueue->usingCounterBasedEvents())
```

Comment:
I think that event should know this instead of going to queue.

Reply:
Agreed, I can keep track of it during the creation of event. However, if it is passed in I would have no way of querying eventpool_desc.pnext from eventpool